### PR TITLE
Move recently added sink and filter which are not part of core into their own packages....

### PR DESCRIPTION
**Pull request by jimlambrt**
I've also added Event.FormattedAs(...) and Event.Format(...) so contributors can write their own filters which set the Event.Formatted map without a data race.

This work is in prep for the new filter node described in ENG-034

ty!

### CODE REVIEW

**Comment by jimlambrt** [comment]
@drewbailey @ncabatoff - I'm not sure who I should ask about reviewing this PR... so I'm starting with you two.  If you'd like to defer to someone else, just let me know.  ty!

**Review by s-christoff — APPROVED** [review]
🤙

File: event.go [hunk 1]
```diff
@@ -27,3 +27,21 @@ type Event struct {
...
+	defer e.l.Unlock()
+	e.Formatted[formatType] = formattedValue
+}
+
+// Format will retrieve the formatted value for the specified format type.  The
+// two value return allows the caller to determine the existence of the format
+// type.
+func (e *Event) Format(formatType string) ([]byte, bool) {
+	e.l.Lock()
+	defer e.l.Unlock()
```
Comment (louisruch) [inline event.go:44]: Since `l` is a RWMutex:
```suggestion
	e.l.RLock()
	defer e.l.RUnlock()
```

**Review by louisruch — APPROVED** [review]
This looks great, one minor comment about using the `RLock`

File: event.go (hunk 1 above)
Comment (jimlambrt) [reply event.go]: ty!  your suggestion led me to look more closely at these two new functions.  I made your suggestion and added a fix for an existing possible runtime panic.  thanks!

File: event.go [hunk 2]
```diff
@@ -33,15 +33,23 @@ type Event struct {
...
 // two value return allows the caller to determine the existence of the format
 // type.
 func (e *Event) Format(formatType string) ([]byte, bool) {
-	e.l.Lock()
-	defer e.l.Unlock()
+	e.l.RLock()
+	defer e.l.RUnlock()
+
+	// while not required, this is a more explicit check for nil
+	if e.Formatted == nil {
```
Comment (louisruch) [inline event.go:50]: good catch

**Review by louisruch — APPROVED** [review]

**Pull request merged**